driver-core/src/main/com/mongodb/client/model/expressions/MqlExpression.java

```diff
@@ -824,8 +820,9 @@ public ArrayExpression<EntryExpression<T>> entrySet() {
     }
 
     @Override
-    public <R extends Expression> MapExpression<R> asMap(final Function<T, EntryExpression<R>> o) {
-        return newMqlExpression(astWrapped("$arrayToObject"));
+    public <R extends Expression> MapExpression<R> asMap(final Function<T, EntryExpression<R>> mapper) {
+        MqlExpression<R> map = (MqlExpression<R>) this.map(mapper);
+        return newMqlExpression(map.astWrapped("$arrayToObject"));
```

Comment:
What's called `map` here is not a map, nor is it a map in [other places in `MqlExpression`](https://github.com/mongodb/mongo-java-driver/pull/1053#discussion_r1067219180) where it's named that way. But here this name especially hurts understanding; it's not helping in other places either, but I did not want to bring it up previously as it was not as egregious as here. `this.map` here produces `ArrayExpression<EntryExpression<? extends R>>` (`? extends R` instead of `R` is here because of the suggestion https://github.com/mongodb/mongo-java-driver/pull/1054#discussion_r1067191394), which means that we should write

```java
@SuppressWarnings("unchecked")
MqlExpression<EntryExpression<? extends R>> array = (MqlExpression<EntryExpression<? extends R>>)this.map(mapper)
return newMqlExpression(array.astWrapped("$arrayToObject"));
```